src/kernel/src/sys/syscall.c

```diff
@@ -0,0 +1,96 @@
+#include "syscall.h"
+
+#include "stdio.h"
+
+// if error, return -1
+int syscall(int sys, ...) {
+    va_list args;
+    va_start(args, sys);
+
+    // variables that can be passed to the SWI
+    int r0 = 0, r1 = 0, r2 = 0, r3 = 0;
+
+    // TODO: the interrupt handler should be able to get the pid from the scheduler?
+    switch (sys) {
+        case SYS_fork:
+            // TODO: should write the pid of the (newly-created) child process where r0 is pointing
+            // (ask the scheduler?)
+
```

Comment:
Looks like a lot of code duplication here

Reply:
I agree. This file is still a draft. We were thinking of rewriting it in assembly, because we would have more control over the stack and registers.